grace/models/classifier.py

```diff
@@ -54,29 +59,24 @@ class GCN(torch.nn.Module):
     def __init__(
         self,
         input_channels: int,
-        hidden_channels: list[int],
+        hidden_graph_channels: list[int],
+        hidden_dense_channels: list[int],
         *,
         dropout: float = 0.0,
         node_output_classes: int = 2,
         edge_output_classes: int = 2,
+        verbose: bool = False,
     ):
-        super(GCN, self).__init__()
+        super(GCNModel, self).__init__()
 
-        # If no hidden channels are specified, train simple Linear classifier:
-        if not hidden_channels:
+        # Define how many (if any) graph conv layers are specified:
+        hidden_channels_list = [
+            input_channels,
+        ]
+        if not hidden_graph_channels:
```

Comment:
I wonder why are we doing this logic? `self.conv_layer_list` and `self.node_dense_list` could be defined as None in the class init and then augmented if `hidden_graph_channels` and `hidden_dense_channels`, this would make the code more succinct. In fact, it looks like before this step `self.conv_layer_list` has not been declared as an attribute in any of the parent or child class. I find it confusing.

Reply:
Makes sense, that's great feedback - refactoring coming in the next commit or two :)